arenadata/Dockerfile.ubuntu

```diff
@@ -29,27 +29,42 @@ RUN set -eux; \
     pip2 install allure-behave==2.4.0; \
     pip2 cache purge
 
+RUN wget https://www.python.org/ftp/python/3.9.19/Python-3.9.19.tgz -O - | tar zxf - && \
+    cd Python-3.9.19 && \
+    ./configure --prefix=/usr/local --enable-shared --enable-optimizations && \
+    make -j6 && \
+    make install && \    
+    echo "/usr/local/lib" >>/etc/ld.so.conf.d/gpdb.conf && \
+	ldconfig && \
+    cd .. && \
+    rm -rf Python-3.9.19.tgz Python-3.9.19
```

Comment:
Why not
```sh
add-apt-repository ppa:deadsnakes/ppa
apt-get update
apt-get install python3.9
```

Reply:
1. python3.9 is not enough, python3.9-dev is required
2. We use now official sources for ubuntu and python
3. Installing python3.9 changes default python in ubuntu which is 3.10 and which potentially can bring problems with OS

Reply:
> python3.9 is not enough, python3.9-dev is required

`apt-get install python3.9-dev`

Reply:
> Installing python3.9 changes default python in ubuntu which is 3.10 and which potentially can bring problems with OS

No:
```bash
python
Python 2.7.18 (default, Oct 15 2023, 16:43:11) 
[GCC 11.4.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 
python3
Python 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
python3.9
Python 3.9.19 (main, Apr  6 2024, 17:57:55) 
[GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

Reply:
I made a variant with installing python 3.9 from deadsnakes, but I still lean to the building python for the container. Actual diff required to make a plpython3 is:

```
diff --git a/README.ubuntu.bash b/README.ubuntu.bash
index de10ef8c7a..17580d2c44 100755
--- a/README.ubuntu.bash
+++ b/README.ubuntu.bash
@@ -1,6 +1,13 @@
 #!/bin/bash
 
 apt-get update
+apt-get install -y software-properties-common
+apt-get update
+DEBIAN_FRONTEND=noninteractive apt-get -y install tzdata
+add-apt-repository ppa:deadsnakes/ppa -y
+apt update
 apt-get install -y \
   bison \
   cmake \
@@ -36,5 +43,8 @@ apt-get install -y \
   pkg-config \
   python2 \
   python2-dev \
+  python3.9 \
+  python3.9-dev \
+  python3.9-distutils \
   rsync \
   zlib1g-dev
```
Here we have to:
1. install software-properties-common to have an add-apt-repository
2. install tzdata separately to avoid interactive install with timezone selection - it's required by python install
3. add deadsnakes repository
4. finally python3.9, python3.9-dev and python3.9-distutils which used by gpdb's configure to get details of python installation.

And, unlike changes to the Dockerfile, all this is going to be done on every developer's machine. I'd object making such a  changes on my workstation without several reason.

Comparing to that, building python on-demand doesn't look a complex procedure, providing much more contol and flexibility in the case if some changes to the build will be needed.

Potentially, we can put a changes above into the Dockerfile layer, but I still vote for building python as it's done with no extra repository and is totally in our hands.

Reply:
> all this is going to be done on every developer's machine

Why do you need to install anything on developers' computers? It's enough to install Docker and do all the development in containers (which is what I do).

Reply:
> > all this is going to be done on every developer's machine
> 
> Why do you need to install anything on developers' computers? It's enough to install Docker and do all the development in containers (which is what I do).

I am following style of the existing container and making as few external dependencies as possible. I am still think that building python on demand is better than using deadsnakes.